### ISSUE TITLE
PadTensors collation function only works when the shape of the data remains unchanged

### ISSUE THREAD

**Issue by robert-osvath** [neuromorphs/tonic#301]
Even though in PyTorch the convention being used for tensors is channel-first, the PadTensors function, as it's implemented right now, can only work if the first dimension is the temporal dimension. In my opinion this makes it unnecessarily complicated, as the data should be already reshaped by the time it gets to padding in the dataloader.

**Comment by biphasic** [comment]
Thanks @robert-osvath , so what format do you have your batches in? Where is the time dimension?

**Comment by robert-osvath** [comment]
@biphasic (channels, depth/time, height, width). I'm not sure if there's a way we could automatically detect the index of the time dimension, but it could easily be fixed with a flag. I would like to open a pull request for this, it would be my first time contributing to an open source project.

**Comment by biphasic** [comment]
Everyone does it differently, some put (time, batch  channels, rest), some put (batch, time, channels, rest). For those two options there exists already a flag. SLAYER for example uses (batch, channels, height, width, time). So I'm not sure there is a common solution

**Comment by robert-osvath** [comment]
Ok, thank you!
